source/AI.cpp

```diff
@@ -1357,6 +1396,22 @@ bool AI::FollowOrders(Ship &ship, Command &command) const
 		else
 			command.SetTurn(TurnToward(ship, TargetAim(ship)));
 	}
+	else if(type == Orders::MINING && targetAsteroid)
+	{
+		ship.SetTargetAsteroid(targetAsteroid);
+		// escorts should chase the player-targeted asteroid
+		MoveToAttack(ship, command, *targetAsteroid);
+	}
+	else if(type == Orders::HARVEST)
+	{
+		if(DoHarvesting(ship, command))
+		{
+			ship.SetCommands(command);
+			ship.SetCommands(firingCommands);
+		}
+		else
+			return false;
+	}
```

Comment:
I don't see why this is here as this PR isn't meant to add a fleet harvests functionality (there is no key for it, and we shouldn't add it in this PR).

Reply:
Not sure if you've played this PR yet but harvesting the flotsam after the fleet breaks it up is a big part of fleet mining.  Take that out and the fleet only shooting the asteroid is a bit useless.

Reply:
I was just not sure what was going on with this, like when and how the harvest command is given.
No I haven't played the PR yet indeed.

Reply:
There's two player flows: with and without asteroid scanner.

With asteroid scanner:

* Press V to target asteroid.  Enables harvest order.
* Press F to attack asteroid.  After asteroid blows up fleet falls back to harvest order.
* With harvest order they chase and collect flotsam.

Without asteroid scanner:

* Pressing V toggles Harvest order on and off.
* Fleet chases after flotsam when harvest order is active.

Harvest orders is also useful for pirate gameplay where your fleet collects flotsam (discarded cargo).

Reply:
I think you should add that clarification to the PR's description.

Reply:
Sure, I can add it as a summary.